test/integration/select_test.lua

```diff
@@ -1581,3 +1583,75 @@ pgroup.test_opts_not_damaged = function(g)
     t.assert_equals(err, nil)
     t.assert_equals(new_select_opts, select_opts)
 end
+
+-- gh-220: bucket_id argument is ignored when it cannot be deduced
+-- from provided select/pairs conditions.
+pgroup.test_select_no_map_reduce = function(g)
+    local customers = helpers.insert_objects(g, 'customers', {
+        {
+            -- bucket_id is 477, storage is s-2
+            id = 1, name = 'Elizabeth', last_name = 'Jackson',
+            age = 12, city = 'New York',
+        }, {
+            -- bucket_id is 401, storage is s-2
+            id = 2, name = 'Mary', last_name = 'Brown',
+            age = 46, city = 'Los Angeles',
+        }, {
+            -- bucket_id is 2804, storage is s-1
+            id = 3, name = 'David', last_name = 'Smith',
+            age = 33, city = 'Los Angeles',
+        }, {
+            -- bucket_id is 1161, storage is s-2
+            id = 4, name = 'William', last_name = 'White',
+            age = 81, city = 'Chicago',
+        },
+    })
+
+    table.sort(customers, function(obj1, obj2) return obj1.id < obj2.id end)
+
+    local storages = {'s1-master', 's1-replica', 's2-master', 's2-replica'}
+    local stat_a = helpers.collect_storage_stat(g.cluster, storages)
```

Comment:
I would replace `stat_a` with `stat_b` to mean "stat before" and "stat after".

Reply:
Re-read your question. Nope, 'a' is just 'a', 'b' is just 'b' and so on: just three points. I can name it stat_1, stat_2 and stat_3 if you find it more preferable.